plugins/pluginmanagers/lz-n.nix

```diff
@@ -0,0 +1,126 @@
+        cmd = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on command.";
+        ft = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on filetype.";
+        colorscheme = helpers.mkNullOrOption maybeRawStrOrList "Lazy-load on colorscheme.";
+        keys = helpers.mkNullOrOption' {
+          type = maybeRawStrOrList;
+          description = "Lazy-load on key mapping. Mode is `n` by default.";
+          example = [
+            "<C-a>"
+            (helpers.mkRaw ''{ "<leader>ft", "<CMD>Neotree toggle<CR>", desc = "NeoTree toggle" }'')
+          ];
+        };
+        priority = helpers.defaultNullOpts.mkUnsignedInt (lib.literalMD "`50` (or `1000` if `colorscheme` is set)") ''
+          Only useful for start plugins (not lazy-loaded) to force loading certain plugins first. 
+        '';
+        load = helpers.mkNullOrLuaFn "Can be used to override the `vim.g.lz_n.load()` function for this plugin.";
+      };
+    };
+in
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "lz-n";
+  originalName = "lz.n";
+  maintainers = [ helpers.maintainers.psfloyd ];
+  defaultPackage = pkgs.vimPlugins.lz-n;
+
+  settingsDescription = ''
+    Options provided to `vim.g.lz_n`.
+
+    `{ load = "fun"; }` -> `vim.g.lz_n = { load = fun, }`
+  '';
+
+  settingsOptions = {
+    load = helpers.defaultNullOpts.mkLuaFn "vim.cmd.packadd" ''
+      Function used by `lz.n` to load plugins.
+    '';
+  };
+
+  callSetup = false; # Does not use setup
+
+  extraOptions = {
+    plugins = mkOption {
+      description = ''
+        List of plugin specs provided to the `require('lz.n').load` function.
+        Plugin specs can be ${helpers.nixvimTypes.rawLua.description}.
+      '';
+      default = [ ];
+      type = types.listOf lzPluginType;
+      example = [
+        {
+          __unkeyed = "neo-tree.nvim";
+          enabled = ''
+            function()
+            return true
+            end
+          '';
+          keys = helpers.mkRaw ''{ "<leader>ft", "<CMD>Neotree toggle<CR>", desc = "NeoTree toggle" }'';
+          after = ''
+            function()
+              require("neo-tree").setup()
+            end
+          '';
+        }
+        {
+          __unkeyed = "telescope.nvim";
+          cmd = [ "Telescope" ];
+          keys = [
+            (helpers.mkRaw ''{ "<leader>fa", "<CMD>Telescope autocommands<CR>", desc = "Telescope autocommands" }'')
+            (helpers.mkRaw ''{ "<leader>fb", "<CMD>Telescope buffers<CR>", desc = "Telescope buffers" }'')
+          ];
+        }
+        {
+          __unkeyed = "onedarker.nvim";
+          colorscheme = [ "onedarker" ];
+        }
+        (helpers.mkRaw ''
+          {
+              "crates.nvim",
+              ft = "toml",
+          },
+        '')
+      ];
+    };
+  };
+
+  extraConfig =
+    cfg:
+    let
+      pluginSpecs = if length cfg.plugins == 1 then head cfg.plugins else cfg.plugins;
+    in
+    {
+      globals.lz_n = cfg.settings;
```

Comment:
It'd be nice if we could copy over any option priorities too (e.g. `mkDefault` or `mkForce`).

[`mkDerivedConfig`](https://noogle.dev/f/lib/modules/mkDerivedConfig) does this for a single option, but here I think we need [`mkAliasAndWrapDefsWithPriority`](https://github.com/NixOS/nixpkgs/blob/585a077fd2b54233843f824f0660599185de1528/lib/modules.nix#L1074-L1078).

That's the function [used by](https://github.com/NixOS/nixpkgs/blob/585a077fd2b54233843f824f0660599185de1528/lib/modules.nix#L1364) `doRename` (which is in-turn used by `mkRenamedOptionModule` and `mkAliasOptionModule`).